hazelcast/src/test/java/com/hazelcast/internal/util/ExceptionUtilTest.java

```diff
@@ -115,6 +119,27 @@ public void testCloneExceptionWithFixedAsyncStackTrace_whenCannotConstructSource
         assertNoAsyncTrace(result);
     }
 
+    @Test
+    public void testRethrowFromCollection_when_notIgnoredThrowableOnList_then_isRethrown() {
+        assertThatExceptionOfType(TestException.class)
+                .isThrownBy(() -> {
+                    rethrowFromCollection(Collections.singleton(new TestException()));
+                });
+        assertThatExceptionOfType(TestException.class)
+                .isThrownBy(() -> {
+                    rethrowFromCollection(Collections.singleton(new TestException()), NullPointerException.class);
+                });
+        assertThatExceptionOfType(TestException.class)
+                .isThrownBy(() -> {
+                    rethrowFromCollection(asList(new NullPointerException(), new TestException()), NullPointerException.class);
+                });
+    }
+
+    @Test
+    public void testRethrowFromCollection_when_ignoredThrowableIsOnlyOnList_then_() throws Throwable {
```

Comment:
is the method name explicitly set like this, because we do nothing if the ignored is the only throwable on the collection?

Reply:
Fixed